pkgs/applications/window-managers/sway/lock-fprintd.nix

```diff
@@ -0,0 +1,43 @@
+{ lib, stdenv, fetchFromGitHub, meson, ninja, pkg-config, scdoc, wayland-scanner
+, wayland, wayland-protocols, libxkbcommon, cairo, gdk-pixbuf, pam
+, dbus, glib, fprintd }:
+
+stdenv.mkDerivation rec {
+  pname = "swaylock-fprintd";
+  version = "unstable-20230130";
+
+  src = fetchFromGitHub {
+    owner = "SL-RU";
+    repo = "swaylock-fprintd";
+    rev = "ffd639a785df0b9f39e9a4d77b7c0d7ba0b8ef79";
+    hash = "sha256-2VklrbolUV00djPt+ngUyU+YMnJLAHhD+CLZD1wH4ww=";
+  };
+
+  postPatch = ''
+    sed -i fingerprint/meson.build \
+      -e 's|/usr/share/dbus-1/interfaces/|'${ fprintd }'/share/dbus-1/interfaces/|'
```

Comment:
Using the `substituteInPlace` helper for this should be better as it prints a warning if there's no match. Submitting a patch with a generic solution to upstream would be even better (I didn't look into it but meson should be able to handle this properly).